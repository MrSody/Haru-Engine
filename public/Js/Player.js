class Player {
    constructor (datos) {
        console.log("datos pj"+ datos.name);
        this.id = datos.id;
        this.name = datos.name;
        this.skinBase = new Image();
        this.skinBase.src = datos.skinBase;
        
		this.pos = {x: datos.posWorld.x, y: datos.posWorld.y};
		
		this.lastFrame = 0;

		////////////////////////

		this.dir = 2;
		this.frame = 0;
		this.path = [[]];
		this.speed = 480;

		//this.sound = new SoundManager();
		this.currhp = 0;
		this.alive = true;
		this.finalDir;
		this.fightFrame = 0;
		this.convPos = 0;
		this.moving = false;
		this.movingDir;
		this.mode = 0, // 0 = normal, 1 = fighting;
		this.hitSpeed = 960;
		this.lastStrike = 0;
		this.moveInterrupt = false;
		this.stepCount = 0;
		this.goAttack = false;
		this.goToNpc = false;
		this.fighting = false;
		this.playerAttacksEnemyID = null;
		this.bgPos = {x: 0, y: 0};
		this.moveAmount = 2;
		this.strength = 50;
		this.gotHit = false;
		this.hitArray = [];
		this.inventory = [];
		this.inventoryMax = 18;
		this.inventoryCol = 0;
		this.inventoryRow = 0;
		this.inventoryBox = 0;
		this.itemSlotsTaken = 0;
		this.lastHitPointUpdate = Date.now();
		this.lastHitUpdate = Date.now();
		this.lastFrameUpdate = Date.now();
		this.conversation = [];
		this.openQuests = [];
		this.closedQuests = [];
		this.pendingQuest = [];
		this.rewardCompleted;
		this.waitForQuestConfirmation = false;
    }

/* ------------------------------ *
    GETTERS
* ------------------------------ */

    getID () {
        return this.id;
    }

    getPos () {
        return this.pos;
    }

    getName () {
		return this.name;
    }

    getDir () {
		return this.dir;
	}

    isMoving () {
		return this.moving;
	}

	isFighting () {
		return this.fighting;
	}

/* ------------------------------ *
    SETTERS
* ------------------------------ */

    setPos (x, y) {
        this.pos.x = x;
        this.pos.y = y;
	}
	
	setDir (dir) {
		this.dir = dir;
	}

/* ------------------------------ *
    FUNCIONES
* ------------------------------ */

	nextFrame () {
		if(this.frame < 3) {
			this.frame++;
		} else {
			this.frame = 0;
		}
	}

}