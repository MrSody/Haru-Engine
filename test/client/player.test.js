/**
 * @jest-environment jsdom
 */

const CONSTANT = require('./constants');
const { createCanvas, loadImage } = require('canvas');

import { response } from 'express';
import Player from '../../public/game/js/modules/entities/player';
import ActionStateEnums from "../../public/game/js/modules/enums/actionState";

test('setPosWorld', () => {
    const player = new Player(CONSTANT.DATA_PLAYER);

    player.setPosWorld(9, 12);

    expect(player.posWorld).toStrictEqual({x: 9, y: 12});
});

describe('nextFrame', () => {
    test('nextFrame - Steps - 0', () => {
        const player = new Player(CONSTANT.DATA_PLAYER);
    
        player.nextFrame();
    
        expect(player.frame).toStrictEqual(1);
    });

    test('nextFrame - Steps - 1', () => {
        const player = new Player(CONSTANT.DATA_PLAYER);
    
        player.nextFrame();
        player.nextFrame();
    
        expect(player.frame).toStrictEqual(2);
    });

    test('nextFrame - Steps - 2', () => {
        const player = new Player(CONSTANT.DATA_PLAYER);
    
        player.nextFrame();
        player.nextFrame();
        player.nextFrame();
    
        expect(player.frame).toStrictEqual(3);
    });

    test('nextFrame - Steps - 4', () => {
        const player = new Player(CONSTANT.DATA_PLAYER);
    
        player.nextFrame();
        player.nextFrame();
        player.nextFrame();
        player.nextFrame();
    
        expect(player.frame).toStrictEqual(0);
    });
});

describe('drawMode', () => {
    test('drawMode - Stand', async () => {
        const player = new Player(CONSTANT.DATA_PLAYER);
        const coordinateX = 160;
        const coordinateY = 160;
    
        const canvaCharacters = createCanvas(544, 352);
        const ctxCharacters = canvaCharacters.getContext('2d');
    
        const image = await loadImage(`${__dirname}\\spritePlayer.png`);
    
        player.skinBase = image;

        player.mode = ActionStateEnums.ActionState().Stand;
        
        player.drawMode(ctxCharacters, coordinateX, coordinateY);
    
        const dataURLCharacters = canvaCharacters.toDataURL();
        expect(dataURLCharacters).toMatchSnapshot();
    });

    test('drawMode - Walking', async () => {
        const player = new Player(CONSTANT.DATA_PLAYER);
        const coordinateX = 160;
        const coordinateY = 160;
    
        const canvaCharacters = createCanvas(544, 352);
        const ctxCharacters = canvaCharacters.getContext('2d');
    
        const image = await loadImage(`${__dirname}\\spritePlayer.png`);
    
        player.skinBase = image;

        player.mode = ActionStateEnums.ActionState().Walking;
        
        player.drawMode(ctxCharacters, coordinateX, coordinateY);
    
        const dataURLCharacters = canvaCharacters.toDataURL();
        expect(dataURLCharacters).toMatchSnapshot();
    });

    test('drawMode - Running', async () => {
        const player = new Player(CONSTANT.DATA_PLAYER);
        const coordinateX = 160;
        const coordinateY = 160;
    
        const canvaCharacters = createCanvas(544, 352);
        const ctxCharacters = canvaCharacters.getContext('2d');
    
        const image = await loadImage(`${__dirname}\\spritePlayer.png`);
    
        player.skinBase = image;

        player.mode = ActionStateEnums.ActionState().Running;
        
        player.drawMode(ctxCharacters, coordinateX, coordinateY);
    
        const dataURLCharacters = canvaCharacters.toDataURL();
        expect(dataURLCharacters).toMatchSnapshot();
    });

    test('drawMode - Fighting', async () => {
        const player = new Player(CONSTANT.DATA_PLAYER);
        const coordinateX = 160;
        const coordinateY = 160;
    
        const canvaCharacters = createCanvas(544, 352);
        const ctxCharacters = canvaCharacters.getContext('2d');
    
        const image = await loadImage(`${__dirname}\\spritePlayer.png`);
    
        player.skinBase = image;

        player.mode = ActionStateEnums.ActionState().Fighting;
        
        player.drawMode(ctxCharacters, coordinateX, coordinateY);
    
        const dataURLCharacters = canvaCharacters.toDataURL();
        expect(dataURLCharacters).toMatchSnapshot();
    });
});