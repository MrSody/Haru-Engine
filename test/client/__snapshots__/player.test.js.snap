// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drawMode drawMode - Fighting 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAiAAAAFgCAYAAACCOcLtAAAABmJLR0QA/wD/AP+gvaeTAAAITElEQVR4nO3dMWhbdx4H8N97VojKbXdDMRRKoSkZbvBQDi0FjwoduhVtJ0opGQPNYKgHDS64odAsgSNQTp0iOrVTtBQEXVQu9Mzd0MLRI8MdIcNdh4vpK62tG9ynyrbs2LKi9xd8PmCQn/7vp/8i+PL//fV/EQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIdlVU+A5TbaitEiPy/bPP390dZi5lHKNn2HAGZRq3oCcFbZZsTw9sbpY25sLzyEAHB+AghLYTJ8rDWbU8fs9PsxvL0hhAAsgbzqCcB5nBQ+nvYeAGkRQEheufpxloCx1mwerII8Za8IANUSQACAhRNAAICFE0AAgIUTQACAhRNASFq5AbU/HMROvz++ftrr/nBgIypA4pwDQvLKX7/0h4PoDwcREdFsrB8as9359YCyZmP9l3u2FzRDAM7LMdJcyCKOYs82IzqtxkGwuN6Jer0exbeDQ2PqV9ejKIrY+VMn+sNBdHrDhRxG5ih2gNlowbAUJlc8joaPo9eOro4AkB4BhKSV+zjK1ks8HB5cz3cjrz2OvPY4snz30HvlWHtAANJlDwjJKlsvERGd3jB+/4ciXv7f3+O3q/uR5fcPjR3tX4v/PsrjX/v/jk5vZ3xftrmYVgwA56N/zYU8qz0gk8evZ/lu/OfRJ7F65dMz3fvoH2/G71b/GKP938ROvx+NZ/hwOntAAGajBUNyjoaP2nOfnzl8RESsXvk0as99Hlm+69kwAIkSQEjKtPCRr9w9d5185a4QApAwAYTklOd+rFz6ZqbwUcpX7sbKpW8O1QQgDQIIyShXPyIi8trjWLl888I1Vy7fjLz2OCLCKghAQgQQklKuVOSX/ja3mmUtqyAA6RBASMLk6kdERL7y9dxqT9ayCgKQBgGEZJQrFFm+e+ycj4vI8vvjw8qsggCkQQAhOVn+ZClqAjA7AYTKHW2//POvg7l/xmRNbRiA6gkgJGGyNTJ48Je515+sqQ0DUD0BhKRsdzbi+gcPIiIezrHsw+sfPIjtzsbTRwKwEAIIlTrafmk21g9ejGIwtw/5pda4dmjDAFRNAKFyU1sieXwytw+YUksbBqBaAghJylZj8MWXF6/zxZcHtS5eCYB5EkBI1r3PIp58/9rM9z/5/rW499kcJwTA3AggVGq0FVG/uj7+vz8cjF9/fC/iu69ejf29d85dd3/vnfjuq1fj43u/XpusXb+6HqOtGSYMwFwIICRlo7MdnVYjImIUEbH21kfx8w9vxN6PH565xt6PH8bPP7wRa299VF4adVqN2Ohsz32+AMxGACE5k0Gh2+3G5Vdej9pLN+PWnYi9n96L0f61Y/eM9q/F3k/vxa07EbWXbsblV16Pbrc7tSYA1RNASM5ke6TVakVRFFEURTx/pRu1F9+P/IX78fa7EbfuHPy9/W5E/sL9qL34fjx/pTse32q1IuJ4mweA6tWqngBEROz0+0/9aWyr1RqHiqPu/Pl8nwVAtayAULnRVkTjxvaZgkGv15vpvdJOvx+NG9s2oAJULKt6Aiy30dbBZtF5KE8mLcNBthlRFMWxcb1e79hKyLRrERH1ev1Qvcn685Bt+g4BzMIKCMkYbZ0tHLRarej1elGv16Ner58YPmatD8CzJ4CQtGltlV6vF+12exwo2u32ieMASJNNqCSt3W5HRIxXOCbDR2m0FZGdMA6ANFkBIWnD2xvRbrfH7ZbTQsXRcZNP2QUgLVZASNpasxnFt83xL2T6w0F0esOpYzutRjQb6+P7DjiADCBFAghLYfKMkJMCSLOx/tSzRABIgxYMALBwAghJu8ippU48BUiXAELSLtJS0Y4BSJcAAgAsnADC0iif43KSsz5PBoDq+RUMSatfXT/0f7fbPfEskG63G412O/z0FiB9VkBIWlEUh/7mPR6AagggJOmkJ+GeV1EU46fgApAOLRiSc1L4mPYcmNK058GUiqKIrF73JFyAhAggLIXTwkfptBACQFqyqifAchttxehZ1J3WNjnrCsZF7j2vbNN3CAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGb2f63691XR3FoqAAAAAElFTkSuQmCC"`;

exports[`drawMode drawMode - Running 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAiAAAAFgCAYAAACCOcLtAAAABmJLR0QA/wD/AP+gvaeTAAAIQUlEQVR4nO3dP2gjVx4H8J/H3lhmD8dlrjnSHGwIB3udmgOTSgTSGaJS6e46F1u4VOniinRZFgI+CKwODJclsKgJESwEdTF3kF18l2AWtshCIIFbolvH0hXe0cqy5D+SrHmGzwcMlubNb6bx8PV7b96LAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAOGmh6BvgetvoRm+e19vNzj6+0Z3PfeR2M39DAJNYKvoG4KJ2s4j2x1tntiln23MPIQBcnv/emMq8ekAGw8ftSmVkm71mMyIiypvzCyF6QAAmc06HNqRlXPg47xgAaRFASF7e+3GRgHG7Uon2x1vnzhUBoFge0wDA3AkgAMDcCSAAwNwJIADA3AkgJC2fgNpst/qv2UbEmb832y0TUQESZyEykpe//dJst6LZbkVERKW8fqLNdv31AmWV8vqrc7bndIcAXJZFlJjKPBYi280i6tXycbD4cz1KpVJ0nrROtCndWo9OpxN7d+vRbLei3mjPZTEyC5EBTEYnNdfCYI/HcPgY/m64dwSA9AggJC2fx5EPvcRBe3zjV8fytuaAAKTLHBCSlQ+9RETUG6+Dx1l7weTDL/3zsvkMxQBwOQIISRpefr1SbkZ58/Wk0j++/6dYyP4bERG97m/im4eP+uHj1Hl2yAVIjgl0TOUqJqGO2/vl4F9/j1/WPonfv/1m3Lz584ljL168Gf8++DlWfvpLvP2HD08c22s2r2yHXJNQASbj4clUZh1ARoWPhexFLN54HIvLdy5U4+h/f42jw3ei173Z/+6qQogAAjAZ0/RIznD4WFp5cOHwERGxuHwnllYexEL2YmRNAIongJCMvPcjl4ePbPHepWtli/dOhRCrowKkw+OYpAz2VCzeeDxR+Mhli/di8cbjkbUBKJYAQhKGez+ypR8uNewyzuLynciWfuh/1gsCkAaPYpIx2EOR3fjnzOoO1tILApAGAYQkHR39I8laAMyGAELhhodfnnz9IN4ofTuz+m+Uvo0nXz/ofzYMA1A8j2GSMDg08p+fvpp5/cGahmEAiieAkJTt+lZ89+PezOt+9+NebNe3zm8IwFwIIBRqePilUl6P/e9nf539749r5wzDABTLI5jCDQ+JfHo/IiIOZniJg1c1x14TgPkSQEhTL1pJ1gJgJgQQ0pTF35KsBcBMCCAUaqMbUbq13v/cbLciImLht9H68tH09b98dFxrsHbE8TVnvTMuABcngJCUrfp21KvliIje/c8jnj6bvNbTZxH3P4+IiF69Wo6t+vZM7hGA6QkgJCcPCp/ej/jm4JOJQsjTZ8fn5pNPhQ+AtAggJGdweKTywUfx1rvP42jl4YXPP1p5GG+9+zwqH3wUEaeHeQAongBCEvaazbHHetlq/LryXhyu7sfR8k70stqINrU4Wt6Jw9X9+HXlvehlqxNdC4D5EEAo3EY3ory5fW4w6C79Lj77IuLl2t14ufY8Dlf343B1P16uPY+Xa3fjsy+O25xlr9mM8ua2CagABVso+ga43ja60ZtVrXxl0jwc7GYRnU7nVLtGoxHVavXc7yIiSqXSiXqD9WdhN/M3BDAJPSAkY6N7sXBQrVaj0WhEqVSKUqk0NnxMWh+AqyeAkLRGozHyu1qt1g8UtVptbDsA0rRU9A3AWWq1WkREv4djMHzkNroRtWx0OwDSZPyaqcxyDsiwfKfc8ubpNTyGh1JG7Wybn3uVwy7mgABMRg8ISbtdqUTnSaX/hkyz3Yp6oz2ybb1ajkp5vX/eMQuQAaRIAOFaeB0oYmwAqZTXT7QDIF0moQIAcyeAkLRpVi214ilAugQQkjbNkIrhGIB0CSAAwNwJIFwb+T4u41xkPxkA0uAtGJJWurV+4vPOzs7YBcZ2dnaiXKuFV28B0qcHhKR1Op0TP7NuD0AxBBCSNG4n3MvqdDojV0kFoFiGYEjOuPAxah+Y3Kj9YHKdTidKWclOuAAJEUC4Fs4KH7mzQggAabGRFlO5qs3oRg2bXLQHY5pzL8tmdAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADCF/wNMWQMtv5jamQAAAABJRU5ErkJggg=="`;

exports[`drawMode drawMode - Stand 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAiAAAAFgCAYAAACCOcLtAAAABmJLR0QA/wD/AP+gvaeTAAAIX0lEQVR4nO3dP2ij5x0H8J/1+uLXXHFu7VICpnAdClfooKVgMoksHatRU2k3Dxk8avTQIVuOg4ALgVOn5ggELaGCg6ChENNCzrh1MAc3JBBISs0p55PVwffKsir5jyRLj+HzAcNJet6f3sXme8/veZ8nAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA4LylRd8At9tBdHvz/L71yC78/CC6c7qTU+uR+R0CmMDyom8Armo9smh/sHXxmM1s7iEEgOvzvzemMq8ZkMHw8aBSGTlmt9mMiIjy5vbcQogZEIDJlBZ9A3Ad48LHZZ8BkBYBhOQVsx9XCRgPKpVof7B16VoRABZLAAEA5k4AAQDmTgABAOZOAAEA5k4AIWnFAtRmu9V/zDYiLvx3s92yEBUgcTYiI3nF0y/Ndiua7VZERFTKG+fGbNfPNiirlDfeXLM9pzsE4LpsosRU5rER2XpkUa+WT4PFH+qR53l09lrnxuT3N6LT6cTuw3o0262oN9pz2YzMRmQAk9GC4VYYnPEYDh/D7w3PjgCQHgGEpBXrOIrWSxy2xw9+81kx1hoQgHRZA0KyitZLRES9cRY8LjoLpmi/FNetNxxOB5AiAYQkDW+/Xik3o7x5tqj0V+/9JpZK/42IiN7JT+LLz572w8fwdU7IBUiPBXRM5SYWoY47++Xwn3+Jl/c+jJ+/83bcvfvDuc+Ojt6Ofx3+EKvf/zHe+eXvzn2222ze2Am5FqECTMYfT6Yy6wAyKnwslY4iu/MsspX3r1Sj++Ofonv8i+id3O2/d1MhRAABmIw/nkzlJgLI4BMtS6WjWF59EqXs0bXqnHR/H69f/vZcCMnvbwggAInwFAzJKGY/CpOGj4iIUvYollefxFLpqP+e3VEB0iGAkJTB1kt259lE4aNQyh5FdufZyNoALJYAQhKGZz9Ky99cec3HRbKV96O0/E3/tVkQgDQIICRjcIaidOcfM6s7WMssCEAaBBCS1O3+NclaAMyGAMLCDbdf9r54Em/lX82s/lv5V7H3xZP+a20YgMUTQEjCYGvk39//beb1B2tqwwAsngBCUrbrW3Hw3e7M6x58txvb9a3LBwIwFwIICzXcfqmUN2L/69l/z/7Xp7UL2jAAiyWAsHDDLZGPHkdExOEMv+LwTc2x3wnAfAkgpKkXrSRrATATAghpKsWfk6wFwEwIICzUQXQjv7/Rf91styIiYumn0fr86fT1P396WmuwdsTNHEwHwNUJICRlq74d9Wo5IqL3+JOI5y8mr/X8RcTjTyIiolevlmOrvj2TewRgegIIySmCwkePI748/HCiEPL8xem1xeJT4QMgLUuLvgFut4Po9qatsR5ZdPZa/ddFe2Q9suh0OrF08p/Ifvx7ZC/fu1K97upn0V35dfRKa5Hn+VmtEd8xg3v3OwQwATMgJGG32Rz7Wa+0Fq9X343jtf3oruxEr1QbMaYW3ZWdOF7bj9er70avtDbRdwEwHwIIC3cQ3Shvbl8aDE6WfxYffxrx6t7DeHXv2zhe24/jtf14de/beHXvYXz86emYi+w2m1He3LYAFWDBTB8zlVm0YArFzqRFOChaMMMajUZUq9VL34uIfgtmVP0Z3bPfIYAJmAEhGQfRvVI4qFar0Wg0Is/zyPN8bPiYtD4AN08AIWmNRmPke7VarR8oarXa2HEApGl50TcAF6nVahER/RmOwfBROIhurNeykeMASJP+NVOZ5RqQYcVJueXN/9/DY7iVMupk2+Lam2y7WAMCMBkzICTtQaUSnb1K/wmZZrsV9UZ75Nh6tRyV8kb/ulM2IANIkQDCrXAWKGJsAKmUN86NAyBdFqECAHMngJC0aXYtteMpQLoEEJI2TUtFOwYgXQIIADB3Agi3RnGOyzhXOU8GgDR4Coak5fc3zr3e2dkZu8HYzs5OlGu18OgtQPrMgJC0Tqdz7mfW4wFYDAGEJI07Cfe6Op3OyF1SAVgsLRiSMy58jDoHpjDqPJhCp9OJPM+chAuQEAGEW+Gi8FG4KIQAkBYHaTGVmzqMblTb5KozGNNce10OowMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAp/A9UiRx6D4sPRAAAAABJRU5ErkJggg=="`;

exports[`drawMode drawMode - Walking 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAiAAAAFgCAYAAACCOcLtAAAABmJLR0QA/wD/AP+gvaeTAAAIlklEQVR4nO3dT4ijZx0H8F8ys2lkKoteinQotmrZyrasPZQcXEgRJfWix/QWEYreVuhhjznOoWAPCqWg5taAFwvCpofCwILESx28dFmxljp7WFDcYhffZjsZD5Nk885kZmfy9xn4fGDZyZvnffJeMnzn+T1/IgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPIKq34Azrne7v5SP6+0efL7vd3lPMdQadN3CGAK66t+ADi10mZ037x+YpNKaXP5IQSAM/PXG7NZ1gjIWPi4UqtNbLLT6UREROXa1vJCiBEQgKkUV/0AcBbHhY9HvQdAWpRgSN9g9OM0AeNKrRbdN5ViAFJnBAQAWDoBBABYOgEEAFg6AQQAWDoBhLQNJqB2utujZbYRceLPne72wZLdR21aBsDKWAVD8oarXzrd7eh0tyMiolap5tpsNR9uUFarVAf3bC3pCQE4K5soMZtlbERW2oxmvXIQLH7WjHK5HNmt7VyT8qVqZFkWO281o9Pdjma7u5xluDYiA5iKEgznwviIx+Hwcfja4dERANIjgJC2wTyOYeml8Mn7UVy/G8X1u1Eo3s+3/biba2sOCEC6zAEhXYPSy+WXsvjG89341tc/jNJGfl7Hfv+V6O+9GP0HL8QHf/zLqPzSrFciIqJpR1SAJAkgpKm0GTu//UVcfvnJWHvs9cHFT480KxRvxFrxRqxdiLjwzLPxzz/fzm3bXqt0bMsOkCAT6JjNIiahDsLH89+/H8W1t89062f/uRqlC9djv78xurbT6SzuhFyTUAGmYg4IaZkhfEREPP6Vm7H+pXdz80MODqizLwhASgQQknP55SenCh9DxbW3Y+3Ch7lrpzlJF4DlEUBIR2kzbv3+52NzPqa39tjrUVy/m7tmFAQgHQIISfnmS1+dW1/FC3/NvTYKApAOAYQ0DM58Ka59MLcus96vjlwzCgKQBgGEZHznh1ejULwxt/42Nj6NW396N3fNKAhAGgQQklEofjb3Pt/7+y9zp+UCkAYBhNUblF8WpXJtKxdClGEAVk8AIQlXarXY7z8+935vf3Tw/+h8mFCGAUiBAEIy9vsbcf/+xbn118u+Hb95JyIiCs12N7aaixtlAeBsBBBW61D5Jbv33bl1/d9/P517XatURz8rwwCslgDCyo2XRP718VNz6/ekvpRhAFZLACEp9+5cjPdvzt7P3udvxL078yvnADBfAgjJeecPERGxPe39/b3XYu/Bc/N6HAAWQABhtXq7Ub5UHb3sdLcPJo4W4ief3Dl7d/291+KL//0o9vsbR94bXwlTvlSN6O2e/QMAmIv1VT8AjLve3IqI61H4WvcfP3014tUfR3zv6unu3fv8jdh78NwofFSubQ3f2m/WK4O+AUhBYdUPwDnX292fuY/SZmS3tnOXhqMirVYr7v6tET+oRjz9VMTFL+dv3e+/Ev29F6P/4IXof/FEbjSl1WpFo9GIiJjc/zxGQEqbvkMAU/DLk9ksIICMwkFpM7IsG11vt9tx871GPPvMwevbH8Vwn4+RVqsV9Xr9YV/l8sO+Jn3G7M/uOwQwBSUYkrDT6TxyaWy9Xs+Fi4iIX/9uus8CYLVMQmX1ertHzms5Trvdnuq9oZ1O52BuiAmoACtl+JjZzKMEMzTcmXQYDg6VYIba7faRkZBJ1yLGSjCT+p8HJRiAqRgBIR293VOFg3q9Hu12O8rlcpTL5WPDx7T9A7B4AghJm1RWabfbB6tbBoGi0Wgc2w6ANJmEStKGy2iHIxy58DHU243GoLxypB0ASVK/ZjbznANy2OCk3LENxcY/90jbw0b3LrLsYg4IwFT88mQ2Cw4gw707hitkOt3taLa7EwNIs16JWqUaEQ9Pu134lusCCMBUlGA4F8b3CGm2uxPb1CrVR+4lAkAaTEIFAJZOACFps+xaasdTgHQJICRtlpKKcgxAugQQAGDpBBDOjdE5Lsc47XkyAKyeVTAkrXypmnvdarWO3WCs1WpFpdGIiONDCgBpMAJC0rIsy/2bd3sAVkMAIU3HnIR7VlmWTdwlFYDVUoIhPceEj4nnwAxNOA9mKMuyKJfKTsIFSIgAwrlwYvgYOiGEAJAW51gwm0WdBTOpbHLaEYxZ7j0rZ8EAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADA9P4P20sTMvq9UksAAAAASUVORK5CYII="`;
